### DISCUSSION TITLE
which hook runs once after --cron

### DISCUSSION THREAD

**Discussion by pkoevesdi** [acmesh-official/acme.sh#5114] [Q&A] [answered]
I have my own deployment script in ansible and I want to run it if any of the certificates are renewed by `--cron`. Like `--notify-mode 0` (bulk mode). But I think, it's not intended to use the notify hook for this, or is it?  
I don't want to run the hook after every renewed domain.

**Comment by neilpang** [comment, marked as answer]
The notify hook is the right tool -- that's exactly what bulk mode is for. In the default bulk mode (`--notify-mode 0`) at the default notify level (`renew`), acme.sh calls the notify hook **once** at the end of `--cron`, and only when at least one cert was renewed or errored -- a run where everything is skipped sends nothing. So it already fires only when something changed.

Set it up with a custom notify hook (a `notify/<name>.sh` script) that triggers your ansible:
```sh
acme.sh --set-notify --notify-hook <yourhook> --notify-mode 0
```
The hook runs once and receives the summary; since it also fires on errors, gate your deploy on success (the subject contains `Success` / the body lists "Successful certs").

If you'd rather keep deployment fully decoupled from notifications, wrap `acme.sh --cron` in your own script and snapshot the cert file mtimes before/after -- `--cron` exits 0 even when nothing renewed, so an mtime comparison is the reliable "did anything change" signal to trigger ansible.

**Comment by pkoevesdi** [comment]
Thank You!
